### ISSUE TITLE
Superfluous brackets added by toString-function

### ISSUE THREAD

**Issue by jvoigtlaender** [elm-lang/core#472]
Consider:
```elm
import Html

type A = A Int

type B = B (List A)

main = Html.text (toString (B [A 1]))
```
The string that gets output is `"B ([A 1])"`, though it could be `"B [A 1]"`.

**Comment by evancz** [comment]
I think this is solved in 0.17. Can you check? Reopen if so, and edit the SSCCE so I can drop it into the new elm-lang.org/try.

**Comment by jvoigtlaender** [comment]
It's not solved in 0.17. I've edited the SSCCEE to be 0.17 compliant.

**Comment by evancz** [comment]
Follow along in the #723 meta issue.
